test/Makefile

```diff
@@ -121,21 +127,28 @@ $(SRC_DIR)/%.o: $(SRC_DIR)/build-cpp-netlib $(SRC_DIR)/build-yaml-cpp $(SRC_DIR)
 	cd $(SRC_DIR) && $(MAKE) $(@:$(SRC_DIR)/%=%)
 
 $(GTEST_SOURCEDIR)/gtest-all.cc $(GTEST_SOURCEDIR)/gtest_main.cc: init-submodules
+$(GMOCK_SOURCEDIR)/gmock-all.cc $(GMOCK_SOURCEDIR)/gmock_main.cc: init-submodules
 
 gtest-all.o: $(GTEST_SOURCEDIR)/gtest-all.cc
 	$(CXX) -c $(CPPFLAGS) -I$(GTEST_DIR) $(CXXFLAGS) -o $@ $^
+gmock-all.o: $(GMOCK_SOURCEDIR)/gmock-all.cc
+	$(CXX) -c $(CPPFLAGS) -I$(GMOCK_DIR) $(CXXFLAGS) -o $@ $^
 
 gtest_main.o: $(GTEST_SOURCEDIR)/gtest_main.cc
 	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c -o $@ $^
+gmock_main.o: $(GMOCK_SOURCEDIR)/gmock_main.cc
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c -o $@ $^
 
 $(GTEST_LIB): gtest-all.o gtest_main.o
 	$(AR) $(ARFLAGS) $@ $^
+$(GMOCK_LIB): gmock-all.o gmock_main.o
```

Comment:
It looks like gmock_main subsumes gtest_main, so if we're linking gmock into everything, we should just be using that. Would it make sense to put both gmock and gtest into a single library that we link, rather than have two?

Reply:
Sounds good, done.